streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -1293,24 +1335,30 @@ public void shouldCommitViaProducerIfEosEnabled() {
             topologyBuilder,
             adminClient,
             stateDirectory,
-            true
+            new StreamsConfig(mkMap(
+                mkEntry(StreamsConfig.APPLICATION_ID_CONFIG, "appId"),
+                mkEntry(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy:1234"),
+                mkEntry(StreamsConfig.PROCESSING_GUARANTEE_CONFIG, eosConfig)
+            ))
         );
         taskManager.setMainConsumer(consumer);
 
         final StateMachineTask task01 = new StateMachineTask(taskId01, taskId01Partitions, true);
-        final Map<TopicPartition, OffsetAndMetadata> offsets = singletonMap(t1p1, new OffsetAndMetadata(0L, null));
-        task01.setCommittableOffsetsAndMetadata(offsets);
+        task01.setCommittableOffsetsAndMetadata(offsetsT01);
         task01.setCommitNeeded();
         taskManager.tasks().put(taskId01, task01);
+        final StateMachineTask task02 = new StateMachineTask(taskId02, taskId02Partitions, true);
+        task02.setCommittableOffsetsAndMetadata(offsetsT02);
+        task02.setCommitNeeded();
+        taskManager.tasks().put(taskId02, task02);
 
-        expect(activeTaskCreator.streamsProducerForTask(taskId01)).andReturn(producer);
-        producer.commitTransaction(offsets);
-        expectLastCall();
-        replay(activeTaskCreator, producer);
+        reset(consumer);
+        expect(consumer.groupMetadata()).andReturn(new ConsumerGroupMetadata("appId")).anyTimes();
+        replay(activeTaskCreator, consumer, producer);
 
         taskManager.commitAll();
 
-        verify(producer);
+        verify(producer, consumer);
```

Comment:
nit: is there a way to verify a function is never called? Like `consumer.commit()`

Reply:
This is verified automatically. As we don't register an expected call, the test would fail if it would be called.